my_enums.rb

```diff
@@ -1,70 +1,159 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+
 
 module Enumerable #:nodoc: all
   def my_each
+    return enum = self.to_enum unless block_given?
     length.times do |n|
       current = self[n]
       yield(current)
     end
-  end
 
   def my_each_with_index
-    length.times do |n|
-      index = n
-      current = self[n]
-      yield(index, current)
+    return enum = self.to_enum unless block_given?
+    my_each do |n|
+      yield(n, self.index(n))
     end
   end
 
   def my_select
     emp_arr = []
+    return enum = self.to_enum unless block_given?
     my_each do |n|
       emp_arr << n if yield(n)
     end
     emp_arr
   end
 
   def my_all?
-    emp_arr = []
-    my_each do |n|
-      emp_arr << n if yield(n)
+    if block_given?
+      begin
+        if yield.is_a? Class
+          my_each do |n|
+            return false if !n.instance_of? yield
+          end
+          true
+        elsif yield.instance_of? Regexp
+          my_each do |n|
+            next if n =~ yield
+            return false
+          end
+          true
+        end
+      rescue => exception
+        my_each do |n|
+          next if yield(n)
+          return false
+        end
+        return true
+      end
+    else
+      return false if self.length == 0
+      my_each do |n|
+        return false if n.nil? || n == false
+      end
+      return true
     end
-    emp_arr == self
   end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class
- [ ] when a Regex is passed as an argument returns true if all of the collection matches the Regex
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern